var/spack/repos/builtin/packages/py-build/package.py

```diff
@@ -12,15 +12,21 @@ class PyBuild(PythonPackage):
     homepage = "https://github.com/pypa/build"
     pypi = "build/build-0.7.0.tar.gz"
 
+    version("0.10.0", sha256="d5b71264afdb5951d6704482aac78de887c80691c52b88a9ad195983ca2c9269")
+    version("0.9.0", sha256="1a07724e891cbd898923145eb7752ee7653674c511378eb9c7691aab1612bc3c")
+    version("0.8.0", sha256="887a6d471c901b1a6e6574ebaeeebb45e5269a79d095fe9a8f88d6614ed2e5f0")
     version("0.7.0", sha256="1aaadcd69338252ade4f7ec1265e1a19184bf916d84c9b7df095f423948cb89f")
 
     variant("virtualenv", default=False, description="Install optional virtualenv dependency")
 
     depends_on("python@3.6:", type=("build", "run"))
-    depends_on("py-setuptools", type="build")
+    depends_on("python@3.7:", when="@0.10:", type=("build", "run"))
+    depends_on("py-flit-core@3.4:", when="@0.10:", type="build")
+    depends_on("py-setuptools", when="@:0.9", type="build")
     depends_on("py-packaging@19:", type=("build", "run"))
-    depends_on("py-pep517@0.9.1:", type=("build", "run"))
+    depends_on("py-pep517@0.9.1:", when="@:0.9", type=("build", "run"))
```

Comment:
Missing dep on py-pyproject-hooks

Reply:
`py-pyproject-hooks` package and dependency added. Thanks to https://github.com/spack/spack/pull/35326/files#diff-60a30aca51f9b417b47bf3689c36fa9d65e469f076e6e1035a28e94bdb10318a for some of the `package.py` (but not using bdist wheel).